connectors/src/connectors/google_drive/index.ts

```diff
@@ -68,825 +67,904 @@ import { terminateAllWorkflowsForConnectorId } from "@connectors/lib/temporal";
     }
-  } catch (err) {
-    logger.error(
+
+    const googleDriveConfigurationBlob = {
+      pdfEnabled: false,
+      largeFilesEnabled: false,
+    };
+
+    const connector = await ConnectorResource.makeNew(
+      "google_drive",
       {
-        err,
+        connectionId: nangoConnectionId,
+        workspaceAPIKey: dataSourceConfig.workspaceAPIKey,
+        workspaceId: dataSourceConfig.workspaceId,
+        dataSourceName: dataSourceConfig.dataSourceName,
       },
-      "Error creating Google Drive connector"
+      googleDriveConfigurationBlob
     );
-    return new Err(new Error("Error creating Google Drive connector"));
-  }
 
-  const googleDriveConfigurationBlob = {
-    pdfEnabled: false,
-    largeFilesEnabled: false,
-  };
-
-  const connector = await ConnectorResource.makeNew(
-    "google_drive",
-    {
-      connectionId: nangoConnectionId,
-      workspaceAPIKey: dataSourceConfig.workspaceAPIKey,
-      workspaceId: dataSourceConfig.workspaceId,
-      dataSourceName: dataSourceConfig.dataSourceName,
-    },
-    googleDriveConfigurationBlob
-  );
+    // We mark it artificially as sync succeeded as google drive is created empty.
+    await syncSucceeded(connector.id);
 
-  // We mark it artificially as sync succeeded as google drive is created empty.
-  await syncSucceeded(connector.id);
+    // We nonetheless launch the incremental sync.
+    const res = await launchGoogleDriveIncrementalSyncWorkflow(connector.id);
+    if (res.isErr()) {
+      return res;
+    }
 
-  // We nonetheless launch the incremental sync.
-  const res = await launchGoogleDriveIncrementalSyncWorkflow(connector.id);
-  if (res.isErr()) {
-    return res;
+    return new Ok(connector.id.toString());
   }
 
-  return new Ok(connector.id.toString());
-}
-
-export async function updateGoogleDriveConnector(
-  connectorId: ModelId,
-  {
+  async update({
     connectionId,
   }: {
-    connectionId?: NangoConnectionId | null;
-  }
-): Promise<Result<string, ConnectorsAPIError>> {
-  const connector = await ConnectorResource.fetchById(connectorId);
-  if (!connector) {
-    logger.error({ connectorId }, "Connector not found");
-    return new Err({
-      message: "Connector not found",
-      type: "connector_not_found",
-    });
-  }
-
-  // Ideally we want to check that the Google Project ID is the same as the one from the connector
-  // I couln't find an easy way to access it from the googleapis library
-  // Workaround is checking the domain of the user who is updating the connector
-  if (connectionId) {
-    try {
-      const oldConnectionId = connector.connectionId;
-      const currentDriveClient = await getDriveClient(oldConnectionId);
-      const currentDriveUser = await currentDriveClient.about.get({
-        fields: "user",
+    connectionId?: string | null;
+  }): Promise<Result<string, ConnectorsAPIError>> {
+    const connector = await ConnectorResource.fetchById(this.connectorId);
```

Comment:
In my opinion, we could take it a step further by implementing this in the `BaseConnectorManager`. Since all the methods involve fetching the connector resource, we could just call `super()` here to have it exposed on the `ConnectorManager`. The main challenge I see is that TypeScript doesn't have a built-in way to enforce that a child class overrides a parent's method (unless the method is abstract) without using TypeScript decorators, which is beyond the scope of this PR.

Reply:
Don't want to do this in this PR